From d894befef6c66718e103a7509c3bdf66c1fcc01e Mon Sep 17 00:00:00 2001
From: You-Sheng Yang <vicamo@gmail.com>
Date: Sat, 17 Feb 2024 22:32:34 +0800
Subject: [PATCH] v4l2: accommodate to struct filed rename of
 vb2_buffer::min_buffers_needed

Signed-off-by: You-Sheng Yang <vicamo@gmail.com>
---
 fthd_v4l2.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/fthd_v4l2.c b/fthd_v4l2.c
index ffd8593..a3974a2 100644
--- a/fthd_v4l2.c
+++ b/fthd_v4l2.c
@@ -683,7 +683,11 @@ int fthd_v4l2_register(struct fthd_private *dev_priv)
 	q->mem_ops = &vb2_dma_sg_memops;
 	q->buf_struct_size = 0;//sizeof(struct vpif_cap_buffer);
 	q->timestamp_flags = V4L2_BUF_FLAG_TIMESTAMP_MONOTONIC;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,8,0)
 	q->min_buffers_needed = 1;
+#else
+	q->min_queued_buffers = 1;
+#endif
 	q->lock = &dev_priv->vb2_queue_lock;
 
 	ret = vb2_queue_init(q);
