From 69ffe34bcca273863655bf2f60909c4c0c94c214 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Fri, 22 Jan 2021 02:25:57 -0800
Subject: [PATCH] cli: add environment variables for flags

Signed-off-by: Christian Stewart <christian@paral.in>
---
 cmd/skiff-core/command_setup.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmd/skiff-core/command_setup.go b/cmd/skiff-core/command_setup.go
index 552d334..c454893 100644
--- a/cmd/skiff-core/command_setup.go
+++ b/cmd/skiff-core/command_setup.go
@@ -20,11 +20,13 @@ var SetupCommands cli.Commands = []cli.Command{
 				Name:        "create-users",
 				Usage:       "If set, core will attempt to create missing users.",
 				Destination: &setupArgs.CreateUsers,
+				EnvVar:      "SKIFF_CORE_CREATE_USERS",
 			},
 			cli.StringFlag{
 				Name:        "work-dir",
 				Usage:       "If set, core will use the directory for working files.",
 				Destination: &setupArgs.WorkDir,
+				EnvVar:      "SKIFF_CORE_WORK_DIR",
 			},
 		},
 		Name:  "setup",
-- 
2.30.0

