From cb316e06346a471df58e4454710dba966e8c4458 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Sat, 12 Mar 2022 17:36:24 -0800
Subject: [PATCH] meson: build: disable wformat-overflow

Signed-off-by: Christian Stewart <christian@paral.in>
---
 meson.build | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/meson.build b/meson.build
index c0cbadecb1..75c169bd35 100644
--- a/meson.build
+++ b/meson.build
@@ -388,6 +388,11 @@ if cc.get_id() == 'gcc'
         possible_common_cc_flags += '-Wno-unused-result'
 endif
 
+# Disable -Wformat-overflow to fix build error
+if cc.get_id() == 'gcc'
+        possible_common_cc_flags += '-Wno-format-overflow'
+endif
+
 # --as-needed and --no-undefined are provided by meson by default,
 # run 'meson configure' to see what is enabled
 possible_link_flags = [
-- 
2.35.1

