From 2afadd13a29f6198fe1a7fdf3ec54888533c661c Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Sat, 20 Aug 2022 23:37:17 -0700
Subject: [PATCH] podman: set no-pivot to default

Signed-off-by: Christian Stewart <christian@paral.in>
---
 cmd/podman/root.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/podman/root.go b/cmd/podman/root.go
index d56aeaad7..9ea3e5cab 100644
--- a/cmd/podman/root.go
+++ b/cmd/podman/root.go
@@ -482,7 +482,7 @@ func rootFlags(cmd *cobra.Command, podmanConfig *entities.PodmanConfig) {
 		noPivotFlagName := "no-pivot"
 		pFlags.BoolVar(
 			&podmanConfig.NoPivot,
-			noPivotFlagName, false,
+			noPivotFlagName, true,
 			"Disable the pivot_root syscall",
 		)
 
-- 
2.40.1

