From 9e11ff8ba311714f2dbb2267ef93df7db4d4deee Mon Sep 17 00:00:00 2001
From: Max Berger <max@berger.name>
Date: Wed, 22 Feb 2023 21:58:14 +0100
Subject: [PATCH] compilation fixes for GCC 12

---
 security/keys/dh.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/security/keys/dh.c b/security/keys/dh.c
index 1abfa70ed6e10..b75b4e6d80239 100644
--- a/security/keys/dh.c
+++ b/security/keys/dh.c
@@ -281,21 +281,21 @@ long __keyctl_dh_compute(struct keyctl_dh_params __user *params,
 
 	memset(&dh_inputs, 0, sizeof(dh_inputs));
 
-	dlen = dh_data_from_key(pcopy.prime, &dh_inputs.p);
+	dlen = dh_data_from_key(pcopy.prime, (void**) &dh_inputs.p);
 	if (dlen < 0) {
 		ret = dlen;
 		goto out1;
 	}
 	dh_inputs.p_size = dlen;
 
-	dlen = dh_data_from_key(pcopy.base, &dh_inputs.g);
+	dlen = dh_data_from_key(pcopy.base, (void**) &dh_inputs.g);
 	if (dlen < 0) {
 		ret = dlen;
 		goto out2;
 	}
 	dh_inputs.g_size = dlen;
 
-	dlen = dh_data_from_key(pcopy.private, &dh_inputs.key);
+	dlen = dh_data_from_key(pcopy.private, (void**) &dh_inputs.key);
 	if (dlen < 0) {
 		ret = dlen;
 		goto out2;
-- 
2.40.1

