From f8e9b2893752c9171ad88735c5330817c0a9cdc3 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Fri, 3 Jun 2016 16:59:21 -0700
Subject: [PATCH 6/6] Apply aufs4 vfs-ino patch

Signed-off-by: Christian Stewart <christian@paral.in>
---
 fs/inode.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/inode.c b/fs/inode.c
index 69b8b52..f46e5c6 100644
--- a/fs/inode.c
+++ b/fs/inode.c
@@ -850,6 +850,8 @@ unsigned int get_next_ino(void)
 	unsigned int *p = &get_cpu_var(last_ino);
 	unsigned int res = *p;
 
+start:
+
 #ifdef CONFIG_SMP
 	if (unlikely((res & (LAST_INO_BATCH-1)) == 0)) {
 		static atomic_t shared_last_ino;
@@ -862,7 +864,7 @@ unsigned int get_next_ino(void)
 	res++;
 	/* get_next_ino should not provide a 0 inode number */
 	if (unlikely(!res))
-		res++;
+		goto start;
 	*p = res;
 	put_cpu_var(last_ino);
 	return res;
-- 
2.7.3

