From 3de34fe2ec7a84c1ba07b71ae3739580da774492 Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei@resin.io>
Date: Mon, 13 Mar 2017 20:32:22 +0100
Subject: [PATCH 3/7] artik710_raptor.h: Boot partition is a fat one

Upstream-Status: Inappropiate [Configuration]

Signed-off-by: Andrei Gherzan <andrei@resin.io>
---
 include/configs/artik710_raptor.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/include/configs/artik710_raptor.h b/include/configs/artik710_raptor.h
index 677bf80..a9a0c0e 100644
--- a/include/configs/artik710_raptor.h
+++ b/include/configs/artik710_raptor.h
@@ -360,14 +360,14 @@
 				"number=0$loop; "			\
 			"else number=$loop; "				\
 			"fi; "						\
-			"ext4load mmc $rootdev:$bootpart $fdtaddr s5p6818-artik710-raptor-rev${number}.dtb && setexpr success 1; " \
+			"fatload mmc $rootdev:$bootpart $fdtaddr s5p6818-artik710-raptor-rev${number}.dtb && setexpr success 1; " \
 			"setexpr loop $loop - 1; "			\
 			"done; "					\
 		"if test $success -eq 0; then "				\
-			"ext4load mmc $rootdev:$bootpart $fdtaddr s5p6818-artik710-raptor-rev00.dtb || "	\
-			"ext4load mmc $rootdev:$bootpart $fdtaddr s5p6818-artik710-raptor.dtb; "	\
+			"fatload mmc $rootdev:$bootpart $fdtaddr s5p6818-artik710-raptor-rev00.dtb || "		\
+			"fatload mmc $rootdev:$bootpart $fdtaddr s5p6818-artik710-raptor.dtb; "		\
 		"fi; "							\
-		"else ext4load mmc $rootdev:$bootpart $fdtaddr $fdtfile; "	\
+		"else fatload mmc $rootdev:$bootpart $fdtaddr $fdtfile; "	\
 		"fi;\0"							\
 	"console=" CONFIG_DEFAULT_CONSOLE				\
 	"consoleon=setenv console=" CONFIG_DEFAULT_CONSOLE		\
@@ -397,8 +397,8 @@
 		"root=/dev/mmcblk${rootdev}p${rootpart} ${root_rw} "	\
 		"rootfstype=${rootfs_type} ${opts} ${recoverymode} "	\
 		"drm_panel=$lcd_panel\0"				\
-	"load_kernel=ext4load mmc ${rootdev}:${bootpart} $kerneladdr $kernel_file\0" \
-	"load_initrd=ext4load mmc ${rootdev}:${bootpart} $ramdiskaddr $ramdisk_file\0" \
+	"load_kernel=fatload mmc ${rootdev}:${bootpart} $kerneladdr $kernel_file\0" \
+	"load_initrd=fatload mmc ${rootdev}:${bootpart} $ramdiskaddr $ramdisk_file\0" \
 	"boot_cmd_initrd="						\
 		"run load_fdt; run load_kernel; run load_initrd;"	\
 		"booti $kerneladdr $ramdiskaddr $fdtaddr\0"		\
-- 
2.7.4

