From 15e21c6c821d8ef98c0f3ad19bee6d0e1d94d875 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Thu, 29 Oct 2020 01:17:16 -0700
Subject: [PATCH] configuration: add skiff core user with sudo

Signed-off-by: Christian Stewart <christian@paral.in>
---
 configuration.nix | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/configuration.nix b/configuration.nix
index 1c66747..fe86c98 100644
--- a/configuration.nix
+++ b/configuration.nix
@@ -70,4 +70,14 @@
   security.sudo.extraRules = [
     { groups = [ "sudo" ]; commands = [ { command = "ALL"; options = [ "NOPASSWD" ]; } ]; }
   ];
+
+  # add skiff core default user
+  users.extraUsers.core = {
+    isNormalUser = true;
+    home = "/home/core";
+    description = "Skiff Core";
+    extraGroups = ["wheel" "vboxusers" "sudo"];
+    createHome = true;
+    shell = "/run/current-system/sw/bin/bash";
+  };
 }
\ No newline at end of file
-- 
2.33.0

